CMakeLists.txt

```diff
@@ -1,88 +1,87 @@
+cmake_minimum_required(VERSION 3.5)
 project(laser_proc)
 
-## Find catkin macros and libraries
-## if COMPONENTS list like find_package(catkin REQUIRED COMPONENTS xyz)
-## is used, also find other catkin packages
-find_package(catkin REQUIRED COMPONENTS roscpp sensor_msgs rosconsole nodelet pluginlib)
-
-###################################################
-## Declare things to be passed to other projects ##
-###################################################
-
-## LIBRARIES: libraries you create in this project that dependent projects also need
-## CATKIN_DEPENDS: catkin_packages dependent projects also need
-## DEPENDS: system dependencies of this project that dependent projects also need
-catkin_package(
-  INCLUDE_DIRS include
-  LIBRARIES laser_proc_library laser_publisher laser_transport laser_proc_ROS LaserProcNodelet
-  CATKIN_DEPENDS roscpp sensor_msgs rosconsole nodelet
-  DEPENDS
+# Default to C++14
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 14)
+endif()
+if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  add_compile_options(-Wall -Wextra -Wpedantic)
+endif()
+
+find_package(ament_cmake REQUIRED)
+find_package(class_loader REQUIRED)
+find_package(rclcpp REQUIRED)
+find_package(sensor_msgs REQUIRED)
+
+add_library(laser_proc
+  src/laser_proc.cpp
+  src/laser_publisher.cpp
+  src/laser_transport.cpp)
+target_include_directories(laser_proc
+  PUBLIC
+  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+  $<INSTALL_INTERFACE:include>
+)
+ament_target_dependencies(laser_proc
+  rclcpp
+  sensor_msgs
 )
-
-###########
-## Build ##
-###########
-
-## Specify additional locations of header files
-include_directories(include ${catkin_INCLUDE_DIRS})
-
-## Library that does actual conversion
-add_library(laser_proc_library src/LaserProc.cpp)
-target_link_libraries(laser_proc_library ${catkin_LIBRARIES})
-
-## Publisher library
-add_library(laser_publisher src/LaserPublisher.cpp)
-target_link_libraries(laser_publisher laser_proc_library ${catkin_LIBRARIES})
-
-## Factory for making LaserPublishers
-add_library(laser_transport src/LaserTransport.cpp)
-target_link_libraries(laser_transport laser_publisher ${catkin_LIBRARIES})
 
 ## ROS Library
-add_library(laser_proc_ROS src/LaserProcROS.cpp)
-target_link_libraries(laser_proc_ROS laser_transport ${catkin_LIBRARIES})
+add_library(laser_proc_ros src/laser_proc_ros.cpp)
+target_link_libraries(laser_proc_ros laser_proc)
+target_include_directories(laser_proc_ros
+  PUBLIC
+  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+  $<INSTALL_INTERFACE:include>
+)
+ament_target_dependencies(laser_proc_ros
+  rclcpp
+  sensor_msgs
+)
 
 ## Nodelet wrapper
-add_library(LaserProcNodelet src/LaserProcNodelet.cpp)
-target_link_libraries(LaserProcNodelet laser_proc_ROS ${catkin_LIBRARIES})
+add_library(laser_proc_nodelet src/laser_proc_nodelet.cpp)
+target_link_libraries(laser_proc_nodelet laser_proc)
+ament_target_dependencies(laser_proc_nodelet
```

Comment:
I think we should get away from the `nodelet` terminology here, as it is a source of confusion.  In ROS 2, we are typically using the term "component", so I'd suggest that (or something like it) instead.

Reply:
I know too little about how laser proc is being used to judge whether we need this in the first place. What does laser proc do when started as a node?

Reply:
I talked about this with @Karsten1987 offline, and it appears that laser_proc subscribes to a `MultiEchoLaserScan` message.  When it gets one, it republishes that data on a different topic, and also picks out the first, last, and most intense scan and publishes them as `LaserScan` messages.

So with all of that being said, I think it makes sense to have this be available as both a regular node and a composable node.  I'm not saying that we necessarily need to do all of that work in this PR, but at least getting away from the nodelet terminology here makes sense to me.